saleor/graphql/order/types.py

```diff
@@ -950,13 +959,17 @@ def resolve_total_balance(root: models.Order, _info):
     @staticmethod
     def resolve_fulfillments(root: models.Order, info):
         def _resolve_fulfillments(fulfillments):
+            requestor = get_user_or_app_from_context(info.context)
             user = info.context.user
-            if user.is_staff:
+            if requestor.has_perm(OrderPermissions.MANAGE_ORDERS):
                 return fulfillments
-            return filter(
-                lambda fulfillment: fulfillment.status != FulfillmentStatus.CANCELED,
-                fulfillments,
-            )
+            elif root.user_id and not user.is_anonymous and root.user_id == user.id:
```

Comment:
```suggestion
            elif root.user_id == user.id and not user.is_anonymous:
```
could be simplified like that, right? we eventually check for `root.user_id == user.id` so it shouldn't matter whether `root.user_id` is None or anything else.